src/psyclone/psyGen.py

```diff
@@ -5156,9 +5190,18 @@ def __init__(self, name, datatype, shape=[], scope='local',
         if not isinstance(shape, list):
             raise TypeError("Symbol shape attribute must be a list.")
 
-        if False in [isinstance(x, (type(None), int)) for x in shape]:
-            raise TypeError("Symbol shape list elements can only be "
-                            "'integer' or 'None'.")
+        for dimension in shape:
+            if isinstance(dimension, Symbol):
+                if dimension.datatype != "integer" or \
+                   len(dimension.shape) > 0:
+                    raise TypeError(
+                        "Symbols that are part of another symbol shape can "
+                        "only be scalar integers, but found '{}'"
```

Comment:
I know it is not required but could you please use {0} rather than {} here for consistency.

Reply:
OK, I will do from now on